File: VB/CustomActionControl.Module/CheckableSimpleAction.vb
Imports System
Imports System.ComponentModel
Imports DevExpress.ExpressApp
Imports DevExpress.ExpressApp.Actions
Imports DevExpress.Persistent.Base

Namespace CustomActionControl.Module
    Public Class CheckableSimpleAction
        Inherits SimpleAction

        Private _Checked As Boolean = False
        Private Sub OnCheckedStateChanged()
            RaiseEvent CheckedStateChanged(Me, EventArgs.Empty)
        End Sub
        Public Sub New(ByVal container As IContainer)
            MyBase.New(container)
        End Sub
        Public Sub New(ByVal owner As Controller, ByVal id As String, ByVal category As PredefinedCategory)
            MyBase.New(owner, id, category)
        End Sub
        Public Property Checked() As Boolean
            Get
                Return _Checked
            End Get
            Set(ByVal value As Boolean)
                If _Checked <> value Then
                    _Checked = value
                    OnCheckedStateChanged()
                    If Enabled.ResultValue AndAlso Active.ResultValue Then
                        DoExecute()
                    End If
                End If
            End Set
        End Property
        Public Event CheckedStateChanged As EventHandler
    End Class
End Namespace

File: CS/EFCore/CustomActionEF/CustomActionEF.Module/CheckableSimpleAction.cs
using System;
using System.ComponentModel;
using DevExpress.ExpressApp;
using DevExpress.ExpressApp.Actions;
using DevExpress.Persistent.Base;

namespace CustomActionControl.Module {
    public class CheckableSimpleAction : SimpleAction {
        private bool _Checked = false;
        private void OnCheckedStateChanged() {
            if(CheckedStateChanged != null) {
                CheckedStateChanged(this, EventArgs.Empty);
            }
        }
        public CheckableSimpleAction(IContainer container) : base(container) { }
        public CheckableSimpleAction(Controller owner, string id, PredefinedCategory category) : base(owner, id, category) { }
        public bool Checked {
            get { return _Checked; }
            set {
                if(_Checked != value) {
                    _Checked = value;
                    OnCheckedStateChanged();
                    if(Enabled.ResultValue && Active.ResultValue) {
                        DoExecute();
                    }
                }
            }
        }
        public event EventHandler CheckedStateChanged;
    }
}
